### ISSUE TITLE
publish event wraps event as payload in a envelope 

### ISSUE THREAD

**Issue by javageek79** [dapr/dapr#1372]
## In what area(s)?

/area runtime
> /area operator
> /area placement
> /area docs
> /area test-and-release

## What version of Dapr?

> 0.3.x

## Expected Behavior

When using `client.publishEvent(topic.getTopicName(), event).doOnError(onError)` where `event` is a Object and conforms to CloudEvent, no extra envelope should be wrapped arround.

Maybe it would be sufficient to have two flavours for publishing. One by only publish the payload, second by publish full fletched event, where no wrapping is necessary.

## Actual Behavior
![image (3)](https://user-images.githubusercontent.com/3714921/77660548-c1b7de00-6f79-11ea-9d9d-916aecb345f8.png)

As can be seen in screenshot, event beeing published by the service is stored in `data` field. It is not possible, to customize type

## Steps to Reproduce the Problem

publish event with DaprClient.

**Comment by artursouza** [comment]
Use Content-Type `application/cloudevents+json` to check if payload is already cloud event and parse it if so.

```
PUT /myresource HTTP/1.1
Host: webhook.example.com
Content-Type: application/cloudevents+json; charset=utf-8
Content-Length: nnnn

{
    "specversion" : "1.0",
    "type" : "com.example.someevent",

    ... further attributes omitted ...

    "data" : {
        ... application data ...
    }
}
```

**Comment by LMWF** [comment]
To summarize, it sounds like there are two asks here, is this correct?
1. Support a publish "overload" that accepts either `CloudEvent` or raw data.  In the first case this should not be rewrapped in an additional `CloudEvent`.  In the latter case, the raw data should be wrapped in `CloudEvent`.
2. Add a correlationid field inside the `CloudEvent`

**Comment by mchmarny** [comment]
#1 from above has a PR [here](https://github.com/dapr/components-contrib/pull/384) which will preserve the published CloudEvent and prevent Dapr from wrapping it into yet another envelope. 

#2 Has already a solution for it. CloudEvent has properties and extensions that can be used to carry that data. Our copy of the CloudEvent (v0.3) struc just doesn't implement these properties today

**Comment by amanbha** [comment]
> #1 from above has a PR [here](https://github.com/dapr/components-contrib/pull/384) which will preserve the published CloudEvent and prevent Dapr from wrapping it into yet another envelope.
> 
> #2 Has already a solution for it. CloudEvent has properties and extensions that can be used to carry that data. Our copy of the CloudEvent (v0.3) struc just doesn't implement these properties today

@javageek79 @artursouza this has been fixed for 0.9.0 release of Dapr, could you please verify your scenario.

**Comment by artursouza** [comment]
@amanbha Thanks.

@javageek79 @RicardoNiepel  Please, try your scenario again with Dapr 0.9.
